lib/client/keystore.go

```diff
@@ -475,12 +475,18 @@ func (fs *FSKeyStore) DeleteKeys() error {
 		return trace.ConvertSystemError(err)
 	}
 	ignoreDirs := map[string]struct{}{tshConfigFileName: {}, tshAzureDirName: {}, tshBin: {}}
+	ignoreFiles := map[string]struct{}{keypaths.VNetClientSSHKey: {}, keypaths.VNetClientSSHKeyPub: {}}
 	for _, file := range files {
 		// Don't delete 'config', 'azure' and 'bin' directories.
 		// TODO: this is hackish and really shouldn't be needed, but fs.KeyDir is `~/.tsh` while it probably should be `~/.tsh/keys` instead.
 		if _, ok := ignoreDirs[file.Name()]; ok && file.IsDir() {
 			continue
 		}
+		// Don't delete VNet client SSH keys on logout in case a user wants to
+		// set these to their own key compatible with their third-party SSH client.
+		if _, ok := ignoreFiles[file.Name()]; ok && !file.IsDir() {
+			continue
+		}
```

Comment:
No reason to bring maps into this. (If you're feeling like it you could also change the check for ignored dirs and get rid of the weird code duplication right after this)

```suggestion
		if !file.IsDir() {
			switch file.Name() {
				case keypaths.VNetClientSSHKey, keypaths.VNetClientSSHKeyPub:
					continue
			}
		}
```